src/node/history.h

```diff
@@ -232,6 +242,76 @@ namespace ccf
         throw std::logic_error("Precondition to mt_retract_to violated");
       mt_retract_to(tree, index);
     }
+
+    std::
+      tuple<std::unique_ptr<hash_vec>, crypto::Sha256Hash, uint64_t, uint32_t>
+      get_path(uint64_t index)
+    {
+      crypto::Sha256Hash res;
+      auto path = std::unique_ptr<hash_vec>(init_path());
+
+      if (!mt_get_path_pre(tree, index, path.get(), res.h))
+        throw std::logic_error("Precondition to mt_get_path violated");
+
+      auto nbelem = mt_get_path(tree, index, path.get(), res.h);
+
+      return {std::move(path), get_root(), index, nbelem};
+    }
+
+    std::vector<uint8_t> get_pathv(uint64_t index)
+    {
+      auto path = get_path(index);
+
+      nlohmann::json j;
```

Comment:
I think that `history.h` shouldn't have to deal with json but only a receipt structure that is json serialiable.

Reply:
Yes, absolutely, there are a bunch of good reasons for that.